docs/machine-learning/tutorials/text-classification-tf.md

```diff
@@ -0,0 +1,260 @@
+
+### Adding context
+
+The [MLContext class](xref:Microsoft.ML.MLContext) is a starting point for all ML.NET operations. Initializing `mlContext` creates a new ML.NET environment that can be shared across the model creation workflow objects. It's similar, conceptually, to `DBContext` in Entity Framework.
+
+1. Replace the `Console.WriteLine("Hello World!")` line in the `Main` method with the following code to declare and initialize the mlContext variable:
+
+   [!code-csharp[CreateMLContext](~/samples/machine-learning/tutorials/TextClassificationTF/Program.cs#CreateMLContext "Create the ML Context")]
+
+1. Add the following as the next line of code in the `Main()` method:
+
+   [!code-csharp[CallReuseAndTuneSentimentModel](~/samples/machine-learning/tutorials/TextClassificationTF/Program.cs#CallReuseAndTuneSentimentModel)]
+
+### Create an intermediate features class
+
+1. Create a utility class called `IntermediateFeatures` to hold intermediate data that will be used by the CustomMapping Estimator in a later step. Add this after the `Main()` method:
+
+[!code-csharp[DeclareIntermediateFeatures](~/samples/machine-learning/tutorials/TextClassificationTF/Program.cs#DeclareIntermediateFeatures)]
+
+## Reuse and tune pre-trained model
+
+1. Create the `ReuseAndTuneSentimentModel()` method, just before the `IntermediateFeatures` utility class, using the following code:
+
+    ```csharp
+    public static ITransformer ReuseAndTuneSentimentModel(MLContext mlContext)
+    {
+
+    }
+    ```
+
+    The `ReuseAndTuneSentimentModel()` method executes the following tasks:
+
+    * Loads the data
+    * Extracts and transforms the data.
+    * Scores the TensorFlow model.
+    * Tunes (retrains) the model.
+    * Returns the model.
+
+1. Create an instance of `IMDBSentiment` called `trainData` and pass it to the `Prediction Engine` by adding the following as the next lines of code in the `ReuseAndTuneSentimentModel()` method:
+
+   [!code-csharp[CreateTrainData](~/samples/machine-learning/tutorials/TextClassificationTF/Program.cs#CreateTrainData)]
+
+### Load the data
+
+Data in ML.NET is represented as an [IDataView class](xref:Microsoft.ML.IDataView). `IDataView` is a flexible, efficient way of describing tabular data (numeric and text). Data can be loaded from a text file or in real time (for example, SQL database or log files) to an `IDataView` object.
+
+1. Load the in-memory collection into an [`IDataView`](xref:Microsoft.ML.IDataView) with the [`LoadFromEnumerable`](xref:Microsoft.ML.DataOperationsCatalog.LoadFromEnumerable%2A) method:
+
+   [!code-csharp[LoadTrainData](~/samples/machine-learning/tutorials/TextClassificationTF/Program.cs#LoadTrainData)]
+
+### Extract and transform the data
+
+1. Create a dictionary to convert words into the integer indexes using the [`LoadFromTextFile`](xref:Microsoft.ML.TextLoaderSaverCatalog.LoadFromTextFile%2A) method to load data from a file, as seen in the following table:
```

Comment:
It's not clear to me why we are performing this step. Can we add a bit more explanation?